react-backend/routes/cbauthentication/checkCBemail.js

```diff
@@ -0,0 +1,39 @@
+const validator = require('validator');
+const getCBAlreadyExists = require('../../database/queries/getCBAlreadyExists');
+const express = require('express');
+const sendresetemail = require('../../functions/sendresetemail');
+
+const router = express.Router();
+
+router.post('/', (req, res, next) => {
+  let body = '';
+  req.on('data', chunk => {
+    body += chunk;
+  });
+
+  req.on('end', () => {
+    const data = JSON.parse(body);
+    if (data.formEmail.length === 0) {
+      res.send('noinput');
+    } else if (validator.isEmail(data.formEmail)) {
+      getCBAlreadyExists(data.formEmail, (error, result) => {
+        if (error) {
+          console.log('error from getCBAlreadyExists ', error);
+          res.status(500).send({
+            error: 'Cannot access database to check if cbemail exists'
+          });
+        } else {
+          res.send(result.rows[0].exists);
+          if (result.rows[0].exists === true) {
+            sendresetemail(data.formEmail);
+          }
+        }
+      });
+    } else if (!validator.isEmail(data.formEmail)) {
+      console.log('This isnt a correct email!?');
+      res.send('email');
+    }
+  });
+});
+
```

Comment:
this file is called `checkCBemail.js`, but I don't feel like that sums it up. This file checks if the CB exists **and** sends a password reset email.

I feel like some kind of naming which pertains to the fact that this is the route which triggers the password email would be more relevant.